app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -118,37 +120,30 @@ public void onDetach() {
         mListener = null;
     }
 
-    // TODO: CHANGE THIS WHEN WE HAVE A FANCY LISTENING INDICATOR
-    private void updateListeningIndicator() {
-        if (isListening)
-            listeningIndicator.setText(R.string.listening_text_indicator);
-        else
-            listeningIndicator.setText(R.string.not_listening_text_indicator);
-    }
-
     public void doListen() {
         Log.d(DEBUG_TAG, "Start listening");
         isListening = true;
-        updateListeningIndicator();
+        listenButton.setImageResource(R.drawable.yes_mic);
         sr.startListening(recognizerIntent);
     }
 
     public void dontListen() {
         Log.d(DEBUG_TAG, "Stop listening.");
         sr.stopListening();
+        listenButton.setImageResource(R.drawable.no_mic);
```

Comment:
I think this code works better the way it was before with a `updateListeningIndicator` function. This function should set the image resource of the button based on isListening. This accomplishes 2 things:

1. As long as we're only updating the listening indicator using this function, we *should* never have a mismatch between whether or not we're listening and what the indicator says.
2. If we want further modifications to be made to the indicator (ie, changing its size @LucyWilcox), this can just be done in the `updateListeningIndicator` function, setting the size based on current volume of speech or something.